### PR TITLE
refactor: tiny adjustment for document testing

### PR DESCRIPTION
**Pull request by arlyxiao**
# Overview

# What is it?

- [ ] Feature / enhancement
- [ ] Bug
- [ ] Docs / tests / types / typos

# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

# Use cases and why

- 1. One use case
- 2. Another use case

# Checklist:

- [x] My code follows the [developer guidelines of this project](https://github.com/QwikDev/qwik/blob/main/CONTRIBUTING.md)
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] Added new tests to cover the fix / functionality

### CODE REVIEW

File: packages/qwik/src/testing/document.ts [hunk 1]
```diff
@@ -20,53 +20,54 @@ export function createDocument(opts?: MockDocumentOptions) {
...
-  return win;
+  return createDocument(opts).defaultView!;
 }
 
 export function ensureGlobals(doc: any, opts?: MockDocumentOptions) {
-  if (!doc[QWIK_DOC]) {
-    if (!doc || doc.nodeType !== 9) {
-      throw new Error(`Invalid document`);
-    }
+  if (!doc || doc.nodeType !== 9) {
```
Comment (gioboa) [inline packages/qwik/src/testing/document.ts:27]: We are changing the behaviour here. If doc[QWIK_DOC] we need to return doc.defaultView

**Review by gioboa — CHANGES_REQUESTED** [review]
Thanks @arlyxiao for your contribution 👏
I added my feedback

File: packages/qwik/src/testing/document.ts [hunk 2]
```diff
@@ -20,53 +20,54 @@ export function createDocument(opts?: MockDocumentOptions) {
...
+    throw new Error(`Invalid document`);
+  }
+
+  if (doc && doc[QWIK_DOC]) {
+    return doc.defaultView;
```
Comment (arlyxiao) [inline packages/qwik/src/testing/document.ts:32]: @gioboa  Here returns doc.defaultView

Comment (gioboa) [reply packages/qwik/src/testing/document.ts:32]: I see but we are throwing an error few line before even if doc[QWIK_DOC] is defined. In the old behaviour we were throwing only if not doc[QWIK_DOC]

Comment (arlyxiao) [reply packages/qwik/src/testing/document.ts:28]: > I see but we are throwing an error few line before even if doc[QWIK_DOC] is defined. In the old behaviour we were throwing only if not doc[QWIK_DOC]

yep. You're right. Updated.

**Review by gioboa — APPROVED** [review]
Thanks @arlyxiao

**Pull request merged**